common/network-common/src/main/java/org/apache/spark/network/util/TransportFrameDecoder.java

```diff
@@ -77,12 +93,37 @@ public void channelRead(ChannelHandlerContext ctx, Object data) throws Exception
         }
         totalSize -= read;
       } else {
-        // Interceptor is not active, so try to decode one frame.
-        ByteBuf frame = decodeNext();
-        if (frame == null) {
-          break;
+        if (isSasl) {
+          if (!isFrameSizeAvailable()) {
```

Comment:
so this means, with sasl, there is no automatic conversion of fetch-to-disk?

Reply:
I think there is. With sasl the pipeline contains two instances of TransportFrameDecoder:  
- the saslFrameDecoder which handles smaller frames  which goes to the saslDecryption
- the "regular" frame decoder where we re doing our trick regarding the bigger fetch chunks

I base this on the transferTo method of EncryptedMessage which [breaks down the original message into smaller chunks when needed](https://github.com/attilapiros/spark/blob/master/common/network-common/src/main/java/org/apache/spark/network/sasl/SaslEncryption.java#L228).